packages/plugin/src/Integrations/PaymentGateways/PayPal/PayPal.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal;
+
+use GuzzleHttp\Client;
+use Solspace\Freeform\Attributes\Integration\Type;
+use Solspace\Freeform\Attributes\Property\Edition;
+use Solspace\Freeform\Attributes\Property\Flag;
+use Solspace\Freeform\Attributes\Property\Input;
+use Solspace\Freeform\Attributes\Property\Validators;
+use Solspace\Freeform\Library\Exceptions\Integrations\IntegrationException;
+use Solspace\Freeform\Library\Integrations\Types\PaymentGateways\PaymentGatewayIntegration;
+
+#[Edition(Edition::PRO)]
+#[Type(
+    name: 'PayPal',
+    type: Type::TYPE_PAYMENT_GATEWAYS,
+    readme: __DIR__.'/README.md',
+    iconPath: __DIR__.'/icon.svg',
+)]
+class PayPal extends PaymentGatewayIntegration
+{
+    public const LOG_CATEGORY = 'PayPal';
+
+    #[Flag(self::FLAG_ENCRYPTED)]
+    #[Flag(self::FLAG_GLOBAL_PROPERTY)]
+    #[Flag(self::FLAG_ENV_SUGGEST)]
+    #[Validators\Required]
+    #[Input\Text(
+        label: 'Client ID',
+        instructions: 'Enter your PayPal REST app client ID.',
+    )]
+    protected string $clientId = '';
+
+    #[Flag(self::FLAG_ENCRYPTED)]
+    #[Flag(self::FLAG_GLOBAL_PROPERTY)]
+    #[Flag(self::FLAG_ENV_SUGGEST)]
+    #[Validators\Required]
+    #[Input\Text(
+        label: 'Client Secret',
+        instructions: 'Enter your PayPal REST app client secret.',
+    )]
+    protected string $clientSecret = '';
+
+    #[Input\Boolean(
+        label: 'Use Sandbox',
+        instructions: 'Enable to use PayPal Sandbox environment.',
+    )]
+    protected bool $useSandbox = true;
+
+    public function getClientId(): ?string
+    {
+        return $this->getProcessedValue($this->clientId);
+    }
+
+    public function getClientSecret(): ?string
+    {
+        return $this->getProcessedValue($this->clientSecret);
+    }
+
+    public function isSandbox(): bool
+    {
+        return (bool) $this->useSandbox;
+    }
+
+    public function checkConnection(Client $client): bool
+    {
+        $token = $this->fetchAccessToken($client);
+
+        return !empty($token);
+    }
```

Comment:
Instead, this should query some endpoint (for instance [webhooks](https://developer.paypal.com/docs/api/webhooks/v1/#webhooks_list)) and see if it returns `HTTP: 200`

```suggestion
    public function checkConnection(Client $client): bool
    {
        $response = $client->get($this->getEndpoint('/v1/notifications/webhooks'), [
            'query' => ['page_size' => 1],
        ]);

        return $response->getStatusCode() === 200;
    }
```